tests/datasource/fluent/test_contexts.py

```diff
@@ -100,35 +109,81 @@ def test_assets_are_persisted_on_creation_and_removed_on_deletion(
 
 
 @pytest.mark.cloud
-@pytest.mark.xfail(reason="Fluent logic attempts to delete before updating")
 def test_context_add_or_update_datasource(
     cloud_api_fake: RequestsMock,
     empty_contexts: CloudDataContext | FileDataContext,
-    db_file: pathlib.Path,
+    # db_file: pathlib.Path, TODO: sqlite deser broken
+    taxi_data_samples_dir: pathlib.Path,
 ):
     context = empty_contexts
 
-    datasource: SqliteDatasource = context.sources.add_sqlite(
-        name="save_ds_test", connection_string=f"sqlite:///{db_file}"
+    datasource = context.sources.add_pandas_filesystem(
+        name="save_ds_test", base_directory=taxi_data_samples_dir
+    )
+    datasource.add_csv_asset(
+        name="my_asset",
     )
 
-    assert datasource.connection_string == f"sqlite:///{db_file}"
-
-    # modify the datasource
-    datasource.connection_string = "sqlite:///"  # type: ignore[assignment]
-    context.sources.add_or_update_sqlite(datasource)
-
+    # TODO: spy the store.delete calls instead of ctx specific tests
     if isinstance(empty_contexts, CloudDataContext):
         # TODO: adjust call counts as needed
         cloud_api_fake.assert_call_count(
             f"{GX_CLOUD_MOCK_BASE_URL}/organizations/{FAKE_ORG_ID}/datasources",
             1,
         )
         cloud_api_fake.assert_call_count(
-            f"{GX_CLOUD_MOCK_BASE_URL}/organizations/{FAKE_ORG_ID}/datasources{datasource.id}",
+            f"{GX_CLOUD_MOCK_BASE_URL}/organizations/{FAKE_ORG_ID}/datasources/{datasource.id}?name={datasource.name}",
             1,
         )
 
+        response = requests.get(
+            f"{GX_CLOUD_MOCK_BASE_URL}/organizations/{FAKE_ORG_ID}/datasources/{datasource.id}"
+        )
+        response.raise_for_status()
+        print(pf(response.json(), depth=4))
+        pytest.xfail(  # TODO: remove this
+            "Immediate persistance for cloud not implemented"
+        )
+        assert response.json()["data"]["attributes"]["datasource_config"].get("assets")
+
+
+@pytest.mark.cloud
+def test_cloud_context_delete_datasource(
```

Comment:
`@pytest.mark.integration`?

Reply:
The whole module is marked as integration tests